### DISCUSSION TITLE
Select component  with custom key passing

### DISCUSSION THREAD

**Discussion by nayeshdaggula** [mantinedev/mantine#3505] [Q&A]
Hi

is there any possible to give option to pass custom key to select component data. 

am having issue when am using country code drop down with select component. most of the country dont have unique value it. because of the error showing in the console. like **required unique key value something** like this. 

if we have option to pass custom key. it may solve that issue?

**any suggestion for how to use the select component in this situation**

**Comment by rtivital** [comment]
Select values in data must be unique. If they are not unique, then it will be impossible to use the component predictably – one value will reference multiple items:
- What label should be displayed as selected in this case?
- What item should be highlighted in dropdown?
- Does it mean that multiple values are selected?

To solve this issue, generate unique value for each item in data array.

**Reply by nayeshdaggula** [reply]
its single selection option only. 

I'm showing the country code with the name in the label, and the country code is in the value; please see the options below.

{label: "+93 (Afghanistan)", value: "93"}

Is it possible to pass a label or another value from an array object instead of a key as value?

**Reply by rtivital** [reply]
`value` must be unique. You can use label as value if it is unique.
